lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -296,9 +305,37 @@ void StagesSubject::stageChanged(
             const TfToken nameToken = changedPath.GetNameToken();
             if (nameToken == UsdGeomTokens->xformOpOrder || UsdGeomXformOp::IsXformOp(nameToken)) {
                 Ufe::Transform3d::notify(ufePath);
+#ifdef UFE_V2_FEATURES_AVAILABLE
+                anyUfeV2NotifSent = true;
+#endif
+            }
+        }
+    }
+
+#ifdef UFE_V2_FEATURES_AVAILABLE
+    // If we get here and didn't send any UFE notif, but yet have paths
+    // then we'll send a catch-all subtree invalidate notification.
+    if (!anyUfeV2NotifSent) {
+        SdfPathVector allPaths = (SdfPathVector)notice.GetResyncedPaths();
+        allPaths.insert(
+            allPaths.end(),
+            notice.GetChangedInfoOnlyPaths().begin(),
+            notice.GetChangedInfoOnlyPaths().end());
+        SdfPath::RemoveDescendentPaths(&allPaths);
+        for (const auto& changedPath : allPaths) {
+            Ufe::Path ufePath;
+            if (changedPath == SdfPath::AbsoluteRootPath()) {
+                ufePath = stagePath(sender);
+            } else {
+                const std::string& usdPrimPathStr = changedPath.GetPrimPath().GetString();
+                ufePath = stagePath(sender) + Ufe::PathSegment(usdPrimPathStr, g_USDRtid, '/');
             }
+
+            auto sceneItem = Ufe::Hierarchy::createItem(ufePath);
+            Ufe::Scene::instance().notify(Ufe::SubtreeInvalidate(sceneItem));
```

Comment:
This feels very awkward.  SubtreeInvalidate is meant to imply a structural change to the hierarchy, but you're adding info-only changes to it.  I don't think we should be doing that.

Reply:
We discussed this internally and since we have no other UFE notif to send this was the only thing we could do. One idea was the creation of a new UFE notif (such as SceneDirty) but we decided against that. @kxl-adsk Can you provide more info?

Reply:
I agree that the catch-all UFE notification for structural change (i.e. ResyncedPaths) should be SubtreeInvalidate --- but isn't that the case already?

For info-only changes, we should be able to extract an attribute out of the notice, and shouldn't need SubtreeInvalidate --- no?

Reply:
Okay I reworked it as we discussed to send Ufe attribute changed notif for the ChangedInfoOnlyPaths. I also incorporated the upcoming UFE changes (.36) to change the notif() method.

I also added the special case when both resync and changed only paths are empty.